flink-queryable-state/flink-queryable-state-runtime/src/test/java/org/apache/flink/queryablestate/itcases/AbstractQueryableStateTestBase.java

```diff
@@ -304,47 +294,17 @@ public Integer getKey(Tuple2<Integer, Long> value) {
 		final JobGraph jobGraph = env.getStreamGraph().getJobGraph();
 		final JobID jobId = jobGraph.getJobID();
 
-		final CompletableFuture<TestingJobManagerMessages.JobStatusIs> failedFuture =
-				notifyWhenJobStatusIs(jobId, JobStatus.FAILED, deadline);
-
-		final CompletableFuture<TestingJobManagerMessages.JobStatusIs> cancellationFuture =
-				notifyWhenJobStatusIs(jobId, JobStatus.CANCELED, deadline);
-
-		cluster.submitJobDetached(jobGraph);
-
-		try {
-			final TestingJobManagerMessages.JobStatusIs jobStatus =
-					failedFuture.get(deadline.timeLeft().toMillis(), TimeUnit.MILLISECONDS);
-
-			assertEquals(JobStatus.FAILED, jobStatus.state());
-		} catch (Exception e) {
-
-			// if the assertion fails, it means that the job was (falsely) not cancelled.
-			// in this case, and given that the mini-cluster is shared with other tests,
-			// we cancel the job and wait for the cancellation so that the resources are freed.
-
-			if (jobId != null) {
-				cluster.getLeaderGateway(deadline.timeLeft())
-						.ask(new JobManagerMessages.CancelJob(jobId), deadline.timeLeft())
-						.mapTo(ClassTag$.MODULE$.<CancellationSuccess>apply(CancellationSuccess.class));
-
-				cancellationFuture.get(deadline.timeLeft().toMillis(), TimeUnit.MILLISECONDS);
-			}
+		clusterClient.submitJob(jobGraph, AbstractQueryableStateTestBase.class.getClassLoader());
 
-			// and we re-throw the exception.
-			throw e;
-		}
+		CompletableFuture<JobResult> jobResultFuture = clusterClient.requestJobResult(jobId);
 
-		// Get the job and check the cause
-		JobManagerMessages.JobFound jobFound = FutureUtils.toJava(
-				cluster.getLeaderGateway(deadline.timeLeft())
-						.ask(new JobManagerMessages.RequestJob(jobId), deadline.timeLeft())
-						.mapTo(ClassTag$.MODULE$.<JobManagerMessages.JobFound>apply(JobManagerMessages.JobFound.class)))
-				.get(deadline.timeLeft().toMillis(), TimeUnit.MILLISECONDS);
+		JobResult jobResult = jobResultFuture.get();
```

Comment:
This should have a timeout. If the timeout is shit we should try to cancel he job.